.github/workflows/issue-types.yml

```diff
@@ -0,0 +1,41 @@
+name: Sync issue types with labels
+
+on:
+  issues:
+    types: [typed,untyped]
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.issue.number }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Update kind labels
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          REPO="${{ github.repository }}"
+          ACTION="${{ github.event.action }}"
+          ISSUE="${{ github.event.issue.number }}"
+
+          if [ "$ACTION" == "typed" ]; then
+            TYPE="${{ github.event.type.name }}"
+            ARGS="--add-label kind/$TYPE"
+          else
+            TYPE=""
+            ARGS=""         
+          fi
+
+          for l in $(gh issue view -R $REPO $ISSUE --json labels -q '.labels[].name' | grep '^kind/'); do
+            if [ "$l" != "kind/$TYPE" ]; then
+              ARGS="$ARGS --remove-label $l"
```

Comment:
I assume that we don't have strange characters in the labels, especially for those with `kind/`. A best practice might be to add quotes to `$l`.

Reply:
Updated, kinda weird to quote within quotes and not really needed, but doesn't hurt